### PR TITLE
Proposal for SH inverter support

### PR DESCRIPTION
**Pull request by benni336**
Hi bohdan,
I would like to propose the following changes to add more support for SH-Inverters.

There is still more todo there, but I guess this would be a good first step to get our two different developments on the same step.
I am currently adapting my other changes to your new OO version (which I prefere over the old one more :+1:  ).

### CODE REVIEW

**Comment by bohdan-s** [comment]
@benni336 can you join a Discord server I just set up? https://discord.gg/7j2MVsT5wn
Might be easier to chat there, by the time you are submitting the code base is already behind. :(

**Comment by bohdan-s** [comment]
Hi, 
You have changed some of the levels if existing registers, did you have some reasons for the change?

**Comment by benni336** [comment]
Joined your channel

File: SunGather/registers.yaml [hunk 1]
```diff
@@ -1,11 +1,31 @@
...
+      datatype: "U16"
+      models: ["SH5K-20", "SH3K6", "SH4K6", "SH5K-V13", "SH5K-30", "SH3K6-30", "SH4K6-30", "SH5.0RS", "SH3.6RS", "SH4.6RS", "SH6.0RS", "SH10RT", "SH8.0RT", "SH6.0RT", "SH5.0RT"] 
+    - name: "dsp_software_version"
+      level: 2
+      address: 4969
+      datatype: "U16"
+      models: ["SH5K-20", "SH3K6", "SH4K6", "SH5K-V13", "SH5K-30", "SH3K6-30", "SH4K6-30", "SH5.0RS", "SH3.6RS", "SH4.6RS", "SH6.0RS", "SH10RT", "SH8.0RT", "SH6.0RT", "SH5.0RT"] 
     - name: "serial_number"
-      level: 3
+      level: 2
```
Comment (benni336) [inline SunGather/registers.yaml:24]: Are you talking about this?

File: SunGather/registers.yaml [hunk 2]
```diff
@@ -1,11 +1,31 @@
...
       address: 4990
       datatype: "UTF-8"
     - name: "device_type_code"
-      level: 3
+      level: 2
```
Comment (benni336) [inline SunGather/registers.yaml:28]: Level change

**Comment by bohdan-s** [comment]
@benni336 sorry, I finally got around to publishing the latest registers.yaml I have been working on.
Could you see if anything is missing? I can see you added a lot in your commit.
If you can re-base it I will approve :) 
The git is now up to date with my current code

**Comment by benni336** [comment]
can be closed as I will do new merge requests

**Pull request closed without merging**